qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -460,267 +860,566 @@ void IndirectFitAnalysisTab::updatePlot(
     const std::string &workspaceName,
     MantidQt::MantidWidgets::PreviewPlot *fitPreviewPlot,
     MantidQt::MantidWidgets::PreviewPlot *diffPreviewPlot) {
-  if (m_fitFunctions == m_propertyFunctions)
+
+  if (previousFitModelSelected())
     IndirectDataAnalysisTab::updatePlot(workspaceName, fitPreviewPlot,
                                         diffPreviewPlot);
   else
     IndirectDataAnalysisTab::updatePlot("", fitPreviewPlot, diffPreviewPlot);
 }
 
+/**
+ * @return  The output workspace name to use for a sequential fit.
+ */
+std::string IndirectFitAnalysisTab::createSequentialFitOutputName() const {
+  return createSingleFitOutputName();
+}
+
+/**
+ * @return The current single fit algorithm for this indirect fit analysis tab.
+ */
+IAlgorithm_sptr IndirectFitAnalysisTab::singleFitAlgorithm() const {
+  auto algorithm = AlgorithmManager::Instance().create("Fit");
+  algorithm->setProperty("WorkspaceIndex",
+                         m_fitPropertyBrowser->workspaceIndex());
+  return algorithm;
+}
+
+/**
+ * @return The current sequential fit algorithm for this indirect fit analysis
+ *         tab.
+ */
+IAlgorithm_sptr IndirectFitAnalysisTab::sequentialFitAlgorithm() const {
+  return singleFitAlgorithm();
+}
+
+/**
+ * Executes the single fit algorithm defined in this indirect fit analysis tab.
+ */
+void IndirectFitAnalysisTab::executeSingleFit() {
+  m_outputFitName = createSingleFitOutputName();
+  runFitAlgorithm(singleFitAlgorithm());
+}
+
+/**
+ * Executes the sequential fit algorithm defined in this indirect fit analysis
+ * tab.
+ */
+void IndirectFitAnalysisTab::executeSequentialFit() {
+  m_outputFitName = createSequentialFitOutputName();
+  runFitAlgorithm(sequentialFitAlgorithm());
+}
+
+/**
+ * @return  The fit function defined in this indirect fit analysis tab.
+ */
+IFunction_sptr IndirectFitAnalysisTab::fitFunction() const {
+  if (!emptyModel())
+    return m_fitPropertyBrowser->getFittingFunction();
+  else
+    return nullptr;
+}
+
+/**
+ * @param function  The function in the fit property browser.
+ * @return          A map from the name of a function in the fit property
+ *                  browser, to the name of a function in the selected model.
+ */
+QHash<QString, QString>
+    IndirectFitAnalysisTab::functionNameChanges(IFunction_sptr) const {
+  return QHash<QString, QString>();
+}
+
+/**
+ * @return  The workspace containing the data to be fit.
+ */
+MatrixWorkspace_sptr IndirectFitAnalysisTab::fitWorkspace() const {
+  return boost::dynamic_pointer_cast<MatrixWorkspace>(
+      m_fitPropertyBrowser->getWorkspace());
+}
+
+/**
+ * Sets the MaxIterations property of the specified algorithm, to the specified
+ * integer value.
+ *
+ * @param fitAlgorithm  The fit algorithm whose MaxIterations property to set.
+ * @param maxIterations The value to set.
+ */
+void IndirectFitAnalysisTab::setMaxIterations(IAlgorithm_sptr fitAlgorithm,
```

Comment:
Why do you need a function for this?

Reply:
IqtFitSequential and IqtFitMultiple's use of IntBoundedValidator means that its MaxIterations property is a long rather than an int. As a result of the property system being unable to perform implicit casts, this produces an error when set. This virtual method is overriden in IqtFit, in which the int is static cast to a long.